src/Module.Server/Rewards/CrpgAgent.cs

```diff
@@ -0,0 +1,22 @@
+﻿namespace Crpg.Module.Rewards;
+internal class CrpgAgent
+{
+    public int CharacterId { get; set; }
+    public int CurrentHealth { get; set; }
+    public int BaseHealthLimit { get; set; }
+    public Dictionary<int, Hitter> Hitters { get; set; }
+    public CrpgAgent(int characterId, int baseHealthLimit)
+    {
+        CharacterId = characterId;
+        CurrentHealth = baseHealthLimit;
+        BaseHealthLimit = baseHealthLimit;
+        Hitters = new();
+    }
+
+    internal class Hitter
+    {
+        public int CharacterId { get; set; }
+        public int TotalDamageDone { get; set; }
+        public bool IsSameTeam { get; set; }
```

Comment:
I thought the feature was about team-hits only. What am I missing 🤔

Reply:
I think Nami wanted to rework the Rating calculation. Because right now the inflictedRatio doesn't consider overkill damage and rewards/punishes you unfairly if let's say you have 1 health remaining but got hit for 20dmg.

This is the line in mind.
float inflictedRatio = MathF.Clamp(blow.InflictedDamage / affectedAgent.BaseHealthLimit, 0f, 1f);

Reply:
> I thought the feature was about team-hits only. What am I missing 🤔

This time this is not a" just in case." This is the pull request that will directly follow this one. We have a bug in rating calculation. If someone deals 300 damage to someone that has 1 / 105  health , it will be counted as 105 damage for rating calculation. as Kadse had to solve the exact same problem for team compensation , i told him to make it so i can reuse it.

Reply:
We don't need to store the hits here for the rating. Also, as always, never include code that might be useful "for later". There is no drawback to add those lines in the following PR.